restore_drosera.sh

```
@@ -506,37 +506,138 @@ source /root/.bashrc 2>/dev/null || true
+        check_status "Применение конфигурации Cadet роли"
     else
-        if [ $attempt -eq 3 ]; then
-            print_color $YELLOW "Предупреждение: не удалось применить конфигурацию трапа"
-            print_color $YELLOW "Возможно, трап уже настроен или есть проблемы с RPC"
+        false # Sets $? to 1 for check_status
+        check_status "Применение конфигурации Cadet роли (НЕ УДАЛОСЬ)"
+        # The script will exit here if check_status is called with $? != 0 due to its internal `exit 1`.
+    fi
+else
+    print_header "НАСТРОЙКА DEFAULT (HELLOWORLD) ТРАПА"
+    cd ~/my-drosera-trap # Ensure correct directory
```

Shellcheck (0.10.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)